portal-impl/src/com/liferay/portal/service/impl/CompanyLocalServiceImpl.java

```diff
@@ -449,6 +459,31 @@ public Company addDBPartitionCompany(
 		}
 	}
 
+	@Override
+	public void afterPropertiesSet() {
```

Comment:
@marianoalvarosaiz @achaparro After a second thought, this clean up is going to cause some performance issue.

For example, I am updating my company on node 1, for cache, it would remove old and add new. Then you are going to receive a notification from cluster to remove your local cache here.

Considering cache performance, it's better to only broadcast CacheRegistryUtil.clear(); when you remove your company.